libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -249,10 +249,20 @@ void AC_PosControl::set_dt(float delta_sec)
 /// To-Do: call this in the main code as part of flight mode initialisation
 void AC_PosControl::set_max_speed_z(float speed_down, float speed_up)
 {
+
```

Comment:
We should try and avoid the duplication to keep the code small but also it actually introduces a bug because a negative speed up or zero speed-down will be accepted I think.

    // update if speed changed at least 1cm/s and positive or non-zero
    if (((fabsf(_speed_down_cms - speed_down) > 1.0f) || (fabsf(_speed_up_cms - speed_up) > 1.0f)) &&
        is_positive(_speed_up_cms) && !is_zero(_speed_down_cms)) {
        _speed_down_cms = -fabsf(speed_down);
        _speed_up_cms = speed_up;
        _flags.recalc_leash_z = true;
        calc_leash_length_z();
    }